packages/runtime/src/interfaces/iocModule.ts

```diff
@@ -0,0 +1,5 @@
+export interface TsoaIocContainer {
+  get<T>(controller: any): T;
+}
+
+export type TsoaIocContainerMethod = (request: any) => TsoaIocContainer;
```

Comment:
```suggestion
export type IocContainerFactory = (request: unknown) => IocContainer;
```

Reply:
I tried typing the request as unknow, but I get
```
Type '(request: e.Request) => IocContainer' is not assignable to type 'IocContainerFactory'.
  Types of parameters 'request' and 'request' are incompatible.
    Type 'unknown' is not assignable to type 'Request<ParamsDictionary>'.
```

Reply:
> I tried typing the request as unknow, but I get
> 
> ```
> Type '(request: e.Request) => IocContainer' is not assignable to type 'IocContainerFactory'.
>   Types of parameters 'request' and 'request' are incompatible.
>     Type 'unknown' is not assignable to type 'Request<ParamsDictionary>'.
> ```

In the editor or with TS itself?
I can type the iocContainer as any one of these just fine:

```
const iocContainer = function (request: e.Request): IocContainer {
const iocContainer: IocContainerFactory = function (request: e.Request): IocContainer {
const iocContainer = function (request: e.Request) {
const iocContainer: IocContainerFactory = function (request: e.Request): IocContainer {
```

Reply:
In vscode. I've committed `(request: unknown)` as the tests passed 🤷‍♂️.